pkg/overlay/pack.go

```diff
@@ -412,34 +406,19 @@ func generateLayer(config types.StackerConfig, _ casext.Engine, mutators []*muta
 		return false, err
 	}
 
-	var ovl overlayMetadata
-	if layer.From.Type != types.BuiltLayer {
-		ovl, err = readOverlayMetadata(config.RootFSDir, name)
-	} else {
-		ovl, err = readOverlayMetadata(config.RootFSDir, layer.From.Tag)
-	}
-	if err != nil {
-		return false, err
-	}
-
 	descs := []ispec.Descriptor{}
 	for i, layerType := range layerTypes {
 		mutator := mutators[i]
 		var desc ispec.Descriptor
 
-		lowerDirs := []string{}
-		for i := len(ovl.Manifests[layerType].Layers) - 1; i >= 0; i-- {
-			lowerDirs = append(lowerDirs, overlayPath(config.RootFSDir, ovl.Manifests[layerType].Layers[i].Digest))
-		}
-
-		blob, mediaType, rootHash, err := generateBlob(layerType, dir, config.OCIDir, lowerDirs)
+		blob, mediaType, rootHash, err := generateBlob(layerType, dir, config.OCIDir)
 		if err != nil {
 			return false, err
 		}
 		defer blob.Close()
 
 		if layerType.Type == "tar" {
-			desc, err = mutator.Add(context.Background(), mediaType, blob, history, mutate.GzipCompressor.WithOpt(gzipBlockSize), nil)
+			desc, err = mutator.Add(context.Background(), mediaType, blob, history, mutate.GzipCompressor, nil)
```

Comment:
Might be useful to leave a breadcrumb pointing to (https://github.com/opencontainers/umoci/pull/509).
Something to give some background on why we got to drop custom umoci fork since umoci picked up the docker-compatible size we used.

Reply:
Either leave a comment in the code here or commit msg.

Reply:
https://github.com/opencontainers/umoci/pull/581 would be the more correct thing to point to, but yes I'll update the commit message.